java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java

```diff
@@ -52,21 +53,33 @@ public void computedFrom(List<SymbolicValue> symbolicValues) {
   }
 
   protected List<ProgramState> copyConstraint(SymbolicValue from, SymbolicValue to, ProgramState programState, BooleanConstraint booleanConstraint) {
-    Constraint constraintLeft = programState.getConstraint(from);
-    if (constraintLeft instanceof BooleanConstraint) {
-      BooleanConstraint boolConstraint = (BooleanConstraint) constraintLeft;
-      return to.setConstraint(programState, shouldNotInverse().equals(booleanConstraint) ? boolConstraint : boolConstraint.inverse());
-    } else if (constraintLeft instanceof ObjectConstraint) {
-      ObjectConstraint objectConstraint = (ObjectConstraint) constraintLeft;
-      if (objectConstraint.isNull()) {
-        return to.setConstraint(programState, shouldNotInverse().equals(booleanConstraint) ? objectConstraint : objectConstraint.inverse());
-      } else if (shouldNotInverse().equals(booleanConstraint)) {
-        return to.setConstraint(programState, objectConstraint);
-      } else if (objectConstraint.isInvalidWith(programState.getConstraint(to))) {
-        return ImmutableList.of();
-      }
+    List<ProgramState> states = new ArrayList<>();
+    states.add(programState);
+    PMap<Class<? extends Constraint>, Constraint> leftConstraints = programState.getConstraints(from);
+    if (leftConstraints != null) {
+      leftConstraints.forEach((d, c) -> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce this lambda expression number of lines from 21 to at most 20. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1188)